ion/tokenizer.go

```diff
@@ -755,6 +755,14 @@ func (t *tokenizer) readRadix(pok, dok matcher) (string, error) {
 	}
 	w.WriteByte(byte(c))
 
+	// At this point we have either 0x or 0b, and it cannot be followed by _
+	nextChar, err2 := t.peek()
+	if err2 != nil {
+		return "", err
+	}
+	if nextChar == '_' {
```

Comment:
Would it make more sense for this `_` check to live inside `readRadixDigits` [just before the `for` loop](https://github.com/amzn/ion-go/blob/74c4cf0a310ebbab78a1b844b055fb08e6aa953c/ion/tokenizer.go#L787) so other calls benefit from the check too?

Reply:
We cannot. [```readDigits()```](https://github.com/amzn/ion-go/blob/74c4cf0a310ebbab78a1b844b055fb08e6aa953c/ion/tokenizer.go#L446) also calls ```readRadixDigits``` function and if we add this ```_``` check in ```readRadixDigits```, then a valid number like ```1_2.3_4``` will be considered as invalid (first digit will be taken in ```readDigits()``` and then when we get to ```readRadixDigits``` the first character is ```_```)

Reply:
Got it, thanks!